server/src/main/java/org/elasticsearch/cluster/coordination/ClusterBootstrapService.java

```diff
@@ -72,15 +72,17 @@ public void start() {
         assert running == false;
         running = true;
 
-        if (initialMasterNodeCount > 0 && transportService.getLocalNode().isMasterNode()) {
+        if ((initialMasterNodeCount > 0 || initialMasterNodes.isEmpty() == false) && transportService.getLocalNode().isMasterNode()) {
             logger.debug("unsafely waiting for discovery of [{}] master-eligible nodes", initialMasterNodeCount);
 
             final ThreadContext threadContext = transportService.getThreadPool().getThreadContext();
             try (ThreadContext.StoredContext ignore = threadContext.stashContext()) {
                 threadContext.markAsSystemContext();
 
                 final GetDiscoveredNodesRequest request = new GetDiscoveredNodesRequest();
-                request.setWaitForNodes(initialMasterNodeCount);
+                if (initialMasterNodeCount > 0) {
```

Comment:
@DaveCTurner I think there's only a UT for this case left here. Before I add one: Is this ok? Simply leaving the default of `1` for the `initalMasterNodeCount` on the request if we have the list?

Reply:
I think we should set both properties (for now).